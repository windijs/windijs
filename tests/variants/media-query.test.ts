import { $dark, $lg, $light, $md, $sm, $xl, $xxl, _lg, _md, _sm, _xl, _xxl, backgroundClip, backgroundClipConfig, backgroundColor, colors, createUtility, dark, landscape, lg, light, ltr, md, motionReduce, motionSafe, portrait, rtl, sm, xl, xxl } from "../../src";

const bg = createUtility("bg")
  .use(backgroundColor(colors))
  .use(backgroundClip(backgroundClipConfig))
  .init();

const utilities = [bg.blue[500], bg.clip.content];

test("mobile first screens", () => {
  expect(sm(...utilities).css).toMatchSnapshot("sm");
  expect(md(...utilities).css).toMatchSnapshot("md");
  expect(lg(...utilities).css).toMatchSnapshot("lg");
  expect(xl(...utilities).css).toMatchSnapshot("xl");
  expect(xxl(...utilities).css).toMatchSnapshot("xxl");
});

test("desktop first screens", () => {
  expect(_sm(...utilities).css).toMatchSnapshot("_sm");
  expect(_md(...utilities).css).toMatchSnapshot("_md");
  expect(_lg(...utilities).css).toMatchSnapshot("_lg");
  expect(_xl(...utilities).css).toMatchSnapshot("_xl");
  expect(_xxl(...utilities).css).toMatchSnapshot("_xxl");
});

test("only screens", () => {
  expect($sm(...utilities).css).toMatchSnapshot("$sm");
  expect($md(...utilities).css).toMatchSnapshot("$md");
  expect($lg(...utilities).css).toMatchSnapshot("$lg");
  expect($xl(...utilities).css).toMatchSnapshot("$xl");
  expect($xxl(...utilities).css).toMatchSnapshot("$xxl");
});

test("motions", () => {
  expect(motionSafe(...utilities).css).toMatchSnapshot("motionSafe");
  expect(motionReduce(...utilities).css).toMatchSnapshot("motionReduce");
});

test("themes", () => {
  expect(dark(...utilities).css).toMatchSnapshot("dark");
  expect(light(...utilities).css).toMatchSnapshot("light");
  expect($dark(...utilities).css).toMatchSnapshot("$dark");
  expect($light(...utilities).css).toMatchSnapshot("$light");
});

test("orientations", () => {
  expect(portrait(...utilities).css).toMatchSnapshot("portrait");
  expect(landscape(...utilities).css).toMatchSnapshot("landscape");
});

test("directions", () => {
  expect(ltr(...utilities).css).toMatchSnapshot("ltr");
  expect(rtl(...utilities).css).toMatchSnapshot("rtl");
});
