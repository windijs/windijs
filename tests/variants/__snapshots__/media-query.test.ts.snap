// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`desktop first screens: _lg 1`] = `
Object {
  "css": Object {
    "@media (max-width: 1024px)": Object {
      "--w-bg-opacity": "1",
      "backgroundClip": "content-box",
      "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
      "webkitBackgroundClip": "content-box",
    },
  },
}
`;

exports[`desktop first screens: _md 1`] = `
Object {
  "css": Object {
    "@media (max-width: 768px)": Object {
      "--w-bg-opacity": "1",
      "backgroundClip": "content-box",
      "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
      "webkitBackgroundClip": "content-box",
    },
  },
}
`;

exports[`desktop first screens: _sm 1`] = `
Object {
  "css": Object {
    "@media (max-width: 640px)": Object {
      "--w-bg-opacity": "1",
      "backgroundClip": "content-box",
      "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
      "webkitBackgroundClip": "content-box",
    },
  },
}
`;

exports[`desktop first screens: _xl 1`] = `
Object {
  "css": Object {
    "@media (max-width: 1280x)": Object {
      "--w-bg-opacity": "1",
      "backgroundClip": "content-box",
      "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
      "webkitBackgroundClip": "content-box",
    },
  },
}
`;

exports[`desktop first screens: _xxl 1`] = `
Object {
  "css": Object {
    "@media (max-width: 1536px)": Object {
      "--w-bg-opacity": "1",
      "backgroundClip": "content-box",
      "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
      "webkitBackgroundClip": "content-box",
    },
  },
}
`;

exports[`directions: ltr 1`] = `
Object {
  "css": Object {
    "[dir='ltr'] &, [dir='ltr']&": Object {
      "--w-bg-opacity": "1",
      "backgroundClip": "content-box",
      "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
      "webkitBackgroundClip": "content-box",
    },
  },
}
`;

exports[`directions: rtl 1`] = `
Object {
  "css": Object {
    "[dir='rtl'] &, [dir='rtl']&": Object {
      "--w-bg-opacity": "1",
      "backgroundClip": "content-box",
      "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
      "webkitBackgroundClip": "content-box",
    },
  },
}
`;

exports[`mobile first screens: lg 1`] = `
Object {
  "css": Object {
    "@media (min-width: 1024px)": Object {
      "--w-bg-opacity": "1",
      "backgroundClip": "content-box",
      "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
      "webkitBackgroundClip": "content-box",
    },
  },
}
`;

exports[`mobile first screens: md 1`] = `
Object {
  "css": Object {
    "@media (min-width: 768px)": Object {
      "--w-bg-opacity": "1",
      "backgroundClip": "content-box",
      "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
      "webkitBackgroundClip": "content-box",
    },
  },
}
`;

exports[`mobile first screens: sm 1`] = `
Object {
  "css": Object {
    "@media (min-width: 640px)": Object {
      "--w-bg-opacity": "1",
      "backgroundClip": "content-box",
      "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
      "webkitBackgroundClip": "content-box",
    },
  },
}
`;

exports[`mobile first screens: xl 1`] = `
Object {
  "css": Object {
    "@media (min-width: 1280x)": Object {
      "--w-bg-opacity": "1",
      "backgroundClip": "content-box",
      "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
      "webkitBackgroundClip": "content-box",
    },
  },
}
`;

exports[`mobile first screens: xxl 1`] = `
Object {
  "css": Object {
    "@media (min-width: 1536px)": Object {
      "--w-bg-opacity": "1",
      "backgroundClip": "content-box",
      "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
      "webkitBackgroundClip": "content-box",
    },
  },
}
`;

exports[`motions: motionReduce 1`] = `
Object {
  "css": Object {
    "@media (prefers-reduced-motion: reduce)": Object {
      "--w-bg-opacity": "1",
      "backgroundClip": "content-box",
      "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
      "webkitBackgroundClip": "content-box",
    },
  },
}
`;

exports[`motions: motionSafe 1`] = `
Object {
  "css": Object {
    "@media (prefers-reduced-motion: no-preference)": Object {
      "--w-bg-opacity": "1",
      "backgroundClip": "content-box",
      "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
      "webkitBackgroundClip": "content-box",
    },
  },
}
`;

exports[`only screens: $lg 1`] = `
Object {
  "css": Object {
    "@media (min-width: 1024px) and (max-width: 1280px)": Object {
      "--w-bg-opacity": "1",
      "backgroundClip": "content-box",
      "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
      "webkitBackgroundClip": "content-box",
    },
  },
}
`;

exports[`only screens: $md 1`] = `
Object {
  "css": Object {
    "@media (min-width: 768px) and (max-width: 1024px)": Object {
      "--w-bg-opacity": "1",
      "backgroundClip": "content-box",
      "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
      "webkitBackgroundClip": "content-box",
    },
  },
}
`;

exports[`only screens: $sm 1`] = `
Object {
  "css": Object {
    "@media (min-width: 640px) and (max-width: 768px)": Object {
      "--w-bg-opacity": "1",
      "backgroundClip": "content-box",
      "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
      "webkitBackgroundClip": "content-box",
    },
  },
}
`;

exports[`only screens: $xl 1`] = `
Object {
  "css": Object {
    "@media (min-width: 1280px) and (max-width: 1536px)": Object {
      "--w-bg-opacity": "1",
      "backgroundClip": "content-box",
      "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
      "webkitBackgroundClip": "content-box",
    },
  },
}
`;

exports[`only screens: $xxl 1`] = `
Object {
  "css": Object {
    "@media (min-width: 1536px)": Object {
      "--w-bg-opacity": "1",
      "backgroundClip": "content-box",
      "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
      "webkitBackgroundClip": "content-box",
    },
  },
}
`;

exports[`orientations: landscape 1`] = `
Object {
  "css": Object {
    "@media (orientation: landscape)": Object {
      "--w-bg-opacity": "1",
      "backgroundClip": "content-box",
      "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
      "webkitBackgroundClip": "content-box",
    },
  },
}
`;

exports[`orientations: portrait 1`] = `
Object {
  "css": Object {
    "@media (orientation: portrait)": Object {
      "--w-bg-opacity": "1",
      "backgroundClip": "content-box",
      "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
      "webkitBackgroundClip": "content-box",
    },
  },
}
`;

exports[`themes: $dark 1`] = `
Object {
  "css": Object {
    ".dark &": Object {
      "--w-bg-opacity": "1",
      "backgroundClip": "content-box",
      "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
      "webkitBackgroundClip": "content-box",
    },
  },
}
`;

exports[`themes: $light 1`] = `
Object {
  "css": Object {
    ".light &": Object {
      "--w-bg-opacity": "1",
      "backgroundClip": "content-box",
      "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
      "webkitBackgroundClip": "content-box",
    },
  },
}
`;

exports[`themes: dark 1`] = `
Object {
  "css": Object {
    "@media (prefers-color-scheme: dark)": Object {
      "--w-bg-opacity": "1",
      "backgroundClip": "content-box",
      "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
      "webkitBackgroundClip": "content-box",
    },
  },
}
`;

exports[`themes: light 1`] = `
Object {
  "css": Object {
    "@media (prefers-color-scheme: light)": Object {
      "--w-bg-opacity": "1",
      "backgroundClip": "content-box",
      "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
      "webkitBackgroundClip": "content-box",
    },
  },
}
`;
